src/Sylius/Behat/Context/Ui/Shop/ProductContext.php

```diff
@@ -351,8 +351,19 @@ public function iShouldSeeTheProductOriginalPrice($price)
         Assert::same($this->showPage->getOriginalPrice(), $price);
     }
 
+    /**
+     * @Then /^I should see ("[^"]+" variant) is not discounted$/
+     */
+    public function iShouldSeeVariantIsNotDiscounted(ProductVariantInterface $variant): void
+    {
+        $this->showPage->selectVariant($variant->getName());
+
+        Assert::same($this->showPage->getPrice(), $this->showPage->getOriginalPrice());
```

Comment:
are you sure about this? from the shop UI perspective if `originalPrice` is not set, the customer sees only `price` not two same value.

Reply:
but its label is still visible, i wasnt able to check if originalPrice is not visible as it always returns true :/ thats the only way i was able to get if the promotion is applied

Reply:
I don't understand why the label is still visible if there is no applied promotion and why it always return true?

Reply:
fixed 💪